### PR TITLE
feature/BBB-186-refactor-createuser

### PR DESCRIPTION
**Pull request by pauljroberts**
…emove usermanagement service dependency

### CODE REVIEW

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/UserManagementClient.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
+package uk.gov.dft.bluebadge.webapp.la.client.usermanagement;
```
Comment (mgilgar) [inline src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/UserManagementClient.java]: Other clients are called BlahBlahApiClient, i.e. UserManagementApiClient

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/CommonResponse.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
+package uk.gov.dft.bluebadge.webapp.la.client.usermanagement.model;
```
Comment (mgilgar) [inline src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/CommonResponse.java]: So, now if we change the model, we have to change in server and client?

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/Password.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.swagger.annotations.ApiModelProperty;
+import java.util.Objects;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+import org.springframework.validation.annotation.Validated;
+
+/** Password */
+@Validated
```
Comment (mgilgar) [inline src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/Password.java]: Do we need @Validated in la-webapp?

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/Password.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    this.password = password;
+    return this;
+  }
+
+  /**
+   * Get password
+   *
+   * @return password
+   */
+  @ApiModelProperty(required = true, value = "")
```
Comment (mgilgar) [inline src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/Password.java]: Do we need this in la-webapp?

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/Password.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    this.passwordConfirm = passwordConfirm;
+    return this;
+  }
+
+  /**
+   * Get passwordConfirm
+   *
+   * @return passwordConfirm
+   */
+  @ApiModelProperty(value = "")
```
Comment (mgilgar) [inline src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/Password.java]: Will we need that in la-webapp? I guess we can keep it, not so sure for the @validated though.

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/User.java [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.swagger.annotations.ApiModelProperty;
+import java.util.Objects;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
+import org.springframework.validation.annotation.Validated;
+
+/** User */
+@Validated
```
Comment (mgilgar) [inline src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/User.java]: Do we need this in la-webapp?

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/User.java [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+   *
+   * @return emailAddress
+   */
+  @ApiModelProperty(
+    example = "rob.worthington@norealserver.com",
+    required = true,
+    value = "email address."
+  )
+  @NotNull
+  @Pattern(
```
Comment (mgilgar) [inline src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/User.java]: Do we need this in la-webapp?

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/User.java [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+    return this;
+  }
+
+  /**
+   * Users name.
+   *
+   * @return name
+   */
+  @ApiModelProperty(example = "Robert Worthington", required = true, value = "Users name.")
+  @NotNull
```
Comment (mgilgar) [inline src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/User.java]: Do we need this in la-webapp?

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/User.java [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+    return this;
+  }
+
+  /**
+   * Id of users role
+   *
+   * @return roleId
+   */
+  @ApiModelProperty(example = "2", required = true, value = "Id of users role")
+  @NotNull
```
Comment (mgilgar) [inline src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/User.java]: Do we need this in la-webapp? (last time I wrote this kind of comment, I think you will get the idea)

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/CommonResponse.java (hunk 2 above)
Comment (pauljroberts) [reply src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/CommonResponse.java]: Yes

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/Password.java (hunk 3 above)
Comment (pauljroberts) [reply src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/Password.java]: No.  But I just left it as generated.

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/Password.java (hunk 4 above)
Comment (pauljroberts) [reply src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/Password.java]: It's used in the client.
   Password passwords = new Password();
    passwords.setPassword(password);
    passwords.setPasswordConfirm(passwordConfirm);

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/Password.java (hunk 5 above)
Comment (pauljroberts) [reply src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/Password.java]: Agree.  Don't need validated.  But have left beans as generated for now.

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/User.java (hunk 7 above)
Comment (pauljroberts) [reply src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/User.java]: Not really.  As above comments.

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/User.java (hunk 9 above)
Comment (pauljroberts) [reply src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/User.java]: Last time I answer too :)  It's up to us.  I'm easy about removing the validation or not.  I have left it all exactly as generated for now.

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/UserManagementClient.java (hunk 1 above)
Comment (pauljroberts) [reply src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/UserManagementClient.java]: Renamed.

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/Password.java (hunk 4 above)
Comment (pauljroberts) [reply src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/Password.java]: Or you just meant the annotation - in which case no, we do not need tha annotation.

File: src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/Password.java (hunk 3 above)
Comment (mgilgar) [reply src/main/java/uk/gov/dft/bluebadge/webapp/la/client/usermanagement/model/Password.java]: Have you run the acceptance tests?

**Pull request merged**